connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoaderTest.java

```diff
@@ -129,4 +133,32 @@ public void testLoadingMixOfValidAndInvalidPlugins() throws Exception {
             assertNotNull(classLoader.pluginClassLoader(pluginClassName));
         }
     }
+
+    @Test
+    public void testAddMultiplePluginsWithSameClass() {
```

Comment:
Is this test buying us much? It looks like it's verifying that multiple versions of the same plugin are recognized by the worker, but not really probing how the worker handles that case.

For what it's worth, I think the functional parts of this change are small and readable enough that a test might not even be warranted.

Reply:
I felt the same but I wanted to see the log statement and there was no way to see it so I added the test, the only other option was to add a ConflictListener but that seemed like a hammer to kill a mosquito :).

Reply:
@C0urante I reworked the tests could you please check now?